src/components/screens/vacation-requests-screen.tsx

```diff
@@ -67,35 +67,35 @@ const VacationRequestsScreen = () => {
    * Fetch vacation requests
    */
   const fetchVacationsRequests = async () => {
-    setLoading(true);
     if (!loggedInUser) return;
-
-    if (!vacationRequests.length) {
-      try {
-        let fetchedVacationRequests: VacationRequest[] = [];
-        if (adminMode) {
-          fetchedVacationRequests = await vacationRequestsApi.listVacationRequests({});
-        } else {
-          fetchedVacationRequests = await vacationRequestsApi.listVacationRequests({
-            userId: loggedInUser?.id
-          });
-        }
-        setVacationRequests(fetchedVacationRequests);
-      } catch (error) {
-        setError(`${strings.vacationRequestError.fetchRequestError}, ${error}`);
+    setLoading(true);
+    try {
+      let fetchedVacationRequests: VacationRequest[];
+      if (adminMode) {
+        fetchedVacationRequests = await vacationRequestsApi.listVacationRequests({});
+      } else {
+        fetchedVacationRequests = await vacationRequestsApi.listVacationRequests({
+          userId: loggedInUser.id
+        });
       }
+      setVacationRequests(fetchedVacationRequests);
+    } catch (error) {
+      setError(`${strings.vacationRequestError.fetchRequestError}, ${error}`);
     }
     setLoading(false);
   };
 
-  useMemo(() => {
+  /**
+   * Re-fetch vacation requests whenever the component is mounted
+   */
```

Comment:
Typically we don't add docs the Use Effects, and I guess while this description is correct, it's also incomplete. I'd suggest removing